### DISCUSSION TITLE
Is there any documentation or guide on creating new UI blocks?

### DISCUSSION THREAD

**Discussion by amilchov** [cortezaproject/corteza-server#1914] [Q&A]
Hello everyone,

I'm looking to create custom UI blocks for my project, but I'm not sure where to begin. Could anyone point me to any existing documentation, tutorials, or guides that explain how to create new UI blocks? Specifically, I am looking for:

- What the general steps are to create a new UI block?
- Any best practices or conventions that should be followed?
- Required dependencies or tools needed for development?
- How to integrate a new UI block into an existing project?

Any help or references would be greatly appreciated!

**Comment by amilchov** [comment]
Hey @kelanik8 , could you provide me with any references or documentation related to my question? Thanks!

**Reply by kelanik8** [reply]
Hello @amilchov you can check this [discuss on the official forum](https://forum.cortezaproject.org/t/custom-ui-blocks/1208)
